Feature/order

jiye-7 opened the pull request:
## 📑 결제하기(주문 페이지)

## 📅 2022.09.04

## 📎 관련 이슈
  resolve #3 

## 💬 작업 내용

- [x] 배송정보에 주소 api 연동
- [x] 주문 상품 정보 recoil로 1개의 상품 저장 후 주문 요약페이지에서도 같은 상태값 사용
- [x] 주문자 정보를 배송정보에서 주문자 정보와 동일 할 때 재사용 할 수 있도록 처리
- [x] 배송 정보에서 우편번호와 주소 검색을 이용하여 배송 정보 등록, 배송 메모 선택 또는 직접 작성하도록 처리
- [x] 결제수단 신용카드 선택시 바로 결제, 무통장 입금 선택시 현금 영수증 신청 여부에 따라 소득공제, 지출증빙 입력하도록 처리
- [x] 결제 관련 체크박스 개별 체크, 전체 체크 기능 추가
---
- [x] 코드 리팩토링(파일 분리 추후 적용)
- [x] 스타일 추후 적용
- [x] 상세페이지에서 받아올 값 상태 관리 수정작업 필요
- [x] 전체 상태값들 중 주문 완료 페이지에서 필요한 값들만 넘겨주도록 상태 관리 수정작업 필요

jiye-7 committed:
build: daum 우편 검색 서비스 추가

jiye-7 committed:
feat: Order, OrderProductInfo;

jiye-7 committed:
feat: Order state, 주문 상품정보, 주문자 정보 레이아웃 추가

jiye-7 committed:
feat: 결제 페이지 전체 레이아웃

jiye-7 committed:
feat: 주소 api 연동

jiye-7 committed:
feat: 주문자 정보/배송 정보 연결

jiye-7 committed:
feat: 주문 상품 정보, 주문자 정보, 배송 정보

jiye-7 committed:
주문 요약 컴포넌트 분리

jiye-7 committed:
feat: 결제 수단

jiye-7 committed:
feat: 주문 기능 구현

주소 api 연동
주문 상품 정보, 주문자 정보, 배송 정보, 주문 요약, 결제수단, 전체 동의 값을 가지고 주문 처리

wjd2676 commented on src/components/Order/style.js:
```diff
@@ -0,0 +1,39 @@
+export const SubTitle = styled.h1`
+  font-size: ${variables.sideSpace.contentSmall};
+`;
+
```
공통 스타일로 적용 하셨네요!! 배워갑니다!

wjd2676 reviewed (commented):
수고 많으셨습니다!

CH4MD0M reviewed (approved):
고생하셨습니다👍

eeoooo commented:
깔끔하게 작성해주셔서 많이 보고 배웠습니다 :) 수고하셨습니다!

jiye-7 committed:
결제 페이지 필수값 체크, 전체 로직 확인

jiye-7 committed:
주문 페이지 컴포넌트 분리

jiye-7 committed:
style: 스타일 적용

jiye-7 committed:
feat: 주문 페이지 기능

- 주문 상품 정보 리코일에 저장한 상품 값 핸들링
- 주문자 정보를 입력받아 처리
- 배송 정보 중 수령인, 연락처는 주문자 정보와 동일하게 처리 할 수 있도록 구현
- 주소 api 를 통한 주소를 입력
- 배송 메모는 선택, 직접입력 가능
- 결제 수단이 무통장일 때 현금 영수증 신청 가능하도록 처리
- 동의/결제까지 필수 정보들이 다 충족되야만 결제요청을 보내도록 처리

jiye-7 committed:
Merge branch 'main' into feature/Order
